src/Datagrid/DatagridMapper.php

```diff
@@ -85,7 +85,7 @@ public function add(
                 array_merge($filterOptions, $fieldDescriptionOptions)
             );
         } else {
-            throw new \RuntimeException(
+            throw new \LogicException(
```

Comment:
IMO, since this exception is about a wrong argument type (when it isn't a instance of `FieldDescriptionInterface` or `string`), [`\TypeError`](https://www.php.net/manual/en/class.typeerror.php) should be used.

Reply:
I'm not familiar with Error, I always work with \Exception.

InvalidArgumentException seems perfect for me: https://www.php.net/manual/fr/class.invalidargumentexception.php#102218

Reply:
Both (`\Error` and `\Exception`) implement `\Throwable`. Before PHP 7, using `\InvalidArgumentException` was the better way to expose these situations. But now we can use the same type used by PHP to tell that an argument type hasn't the expected type: https://3v4l.org/065YF.

Reply:
\RuntimeException was mean to be catch by the developper, and \LogicException.
What about \Error ?

If we throw an Error and annotate the Phpdoc. Isn't like saying you have to try/catch the `Throwable` ? Or Error are not meant to be cought  ?

Reply:
`\TypeError` and not meant to be caught I would say, although I've seen it in a project I took over :sweat: . You can witness one in action here: https://3v4l.org/8l3ol
What would be great with using that is that it will be forward compatible with union types. BTW, the phpdoc on tha tmethod looks wrong.

Reply:
@VincentLanglet are you going to use `\TypeError` or should we mark this as resolved?